src/compiler/utilities.rs

```diff
@@ -761,8 +763,15 @@ fn create_file_diagnostic(
     start: isize,
     length: isize,
     message: &DiagnosticMessage,
+    args: Option<Vec<String>>,
 ) -> DiagnosticWithLocation {
-    let text = get_locale_specific_message(message);
+    let mut text = get_locale_specific_message(message);
+
+    if let Some(args) = args {
+        if !args.is_empty() {
+            text = format_string_from_args(&text, args);
+        }
+    }
```

Comment:
Just playing a little type tetris here, this also works to get rid of the nested if-statements (but is worse in other ways):

```rs
let args: Vec<_> = args.into_iter().flatten().collect();

if !args.is_empty() {
    text = format_string_from_args(&text, args);
}
```